### PR TITLE
React 05

### PR DESCRIPTION
**Pull request by Maaz-Mustafa**
Completed React Module 05 and added Sorting and Searching
Screenshot from 2023-01-05 11-21-38

### CODE REVIEW

File: fetch-todolist/src/component/AddToDoList.js [hunk 1]
```diff
@@ -1,51 +1,67 @@
...
     setDdate(e.target.value);
   };
+  const activity = {
+    id: "",
+    task: task,
+    dueDtae: ddate,
+    status: false,
+  };
+  const handleAdd = () => {
+    fetch("http://localhost:8080/todo", {
```
Comment (VrushabhB) [inline fetch-todolist/src/component/AddToDoList.js:21]: move api calls to separate files

File: fetch-todolist/src/component/Filters.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import {memo} from 'react';
+const FiltersComponent = ({handleStatusFilter,searchTask,handleOrder}) => {
+    
+    
+    
+    return ( <>
```
Comment (VrushabhB) [inline fetch-todolist/src/component/Filters.jsx:6]: create common select component

File: fetch-todolist/src/component/Home.js [hunk 3]
```diff
@@ -26,51 +29,61 @@ const Home = () => {
...
+    setTodos(completedTask);
+  };
+  const searchTask = (search) => {
+    let searchedList = list;
+    searchedList = searchedList.filter((item) => {
+      return item.task.includes(search);
+    });
+    setTodos(searchedList);
+  };
+  useEffect(() => {
```
Comment (VrushabhB) [inline fetch-todolist/src/component/Home.js:60]: while writing the code maintain order of function type all use effect should be at end

File: fetch-todolist/src/component/Task.jsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { Link, useParams } from "react-router-dom";
+
+const Task = () => {
```
Comment (VrushabhB) [inline fetch-todolist/src/component/Task.jsx:3]: use propType for all the component

File: fetch-todolist/src/component/AddToDoList.js (hunk 1 above)
Comment (Maaz-Mustafa) [reply fetch-todolist/src/component/AddToDoList.js:21]: I have created a separate folder name services, in that I've made two files for api.

File: fetch-todolist/src/component/Filters.jsx (hunk 2 above)
Comment (Maaz-Mustafa) [reply fetch-todolist/src/component/Filters.jsx:6]: created a component named SelectBox and using it by passing arrays to it, in the latest commit.

File: fetch-todolist/src/component/Home.js (hunk 3 above)
Comment (Maaz-Mustafa) [reply fetch-todolist/src/component/Home.js:60]: Done

**Pull request merged**